libnymea-core/integrations/thingmanagerimplementation.cpp

```diff
@@ -1525,27 +1525,30 @@ void ThingManagerImplementation::loadConfiguredThings()
     foreach (const QString &idString, settings.childGroups()) {
         settings.beginGroup(idString);
         QString thingName = settings.value("thingName").toString();
-        if (!settings.contains("thingName")) { // nymea < 0.20
-            thingName = settings.value("devicename").toString();
-        }
         PluginId pluginId = PluginId(settings.value("pluginid").toString());
+        ThingClassId thingClassId = ThingClassId(settings.value("thingClassId").toString());
+
+        if (m_supportedThings.contains(thingClassId) && m_supportedThings.value(thingClassId).pluginId() != pluginId) {
+            PluginId newPluginId = m_supportedThings.value(thingClassId).pluginId();
+            qCInfo(dcThingManager()) << "Thing class" << thingClassId << "has moved from plugin" << pluginId << "to" << newPluginId;
+            pluginId = newPluginId;
+            settings.setValue("pluginid", pluginId);
```

Comment:
If the plugin ID can change, do we even need to store it? Since the `thingClassId` is already unique and we verify if we have a plugin for that thingclass, the information is already available after loading the plugin.

Reply:
The reason is to be able to load the thingclass from the cache in case a plugin fails to load... Currently the caches are based on the plugin id. That could be changed, but is a bit more work.